scripts/biomedical/VirtualMetabolicHuman/format_metabolite.py

```diff
@@ -205,120 +111,156 @@ def add_query_result(df, col, dcid):
+    pre-existing matching kegg ids on dc
+    Args:
+        arr_hmdb = array with kegg ids to query
+    Returns:
+        result = result of dc query
+    """
     query_str = """
     SELECT DISTINCT ?drug ?id
     WHERE {{
     ?drug typeOf ChemicalCompound .
-    ?drug inChIKey ?id .
-    ?drug inChIKey {0} .
+    ?drug keggCompoundID ?id .
+    ?drug keggCompoundID {0} .
     }}
-    """.format(arr_inchi_3)
+    """.format(arr_kegg)
     result = dc.query(query_str)
-    dcid_inch = clean_result(result)
-    df = add_query_result(df, "inchiKey", dcid_inch)
-    #inchi match complete (3)
+    return result
 
-    #hmdb matches with dc
-    list_hmdb = df[['hmdb']].T.stack().tolist()
-    list_hmdb_1 = list_hmdb[0:1000]
-    list_hmdb_2 = list_hmdb[1000:1513]
-    arr_hmdb_1 = np.array(list_hmdb_1)
-    arr_hmdb_2 = np.array(list_hmdb_2)
+def chebi_query(arr_chebi):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching chebi ids on dc
+    Args:
+        arr_hmdb = array with chebi ids to query
+    Returns:
+        result = result of dc query
+    """
     query_str = """
     SELECT DISTINCT ?drug ?id
     WHERE {{
     ?drug typeOf ChemicalCompound .
-    ?drug humanMetabolomeDatabaseID ?id .
-    ?drug humanMetabolomeDatabaseID {0} .
+    ?drug chebiID ?id .
+    ?drug chebiID {0} .
     }}
-    """.format(arr_hmdb_1)
+    """.format(arr_chebi)
     result = dc.query(query_str)
-    dcid_hmdb = clean_result(result)
-    df = add_query_result(df, "hmdb", dcid_hmdb)
+    return result
 
+def drugbank_query(arr_drug):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching drugbank ids on dc
+    Args:
+        arr_hmdb = array with drugbank ids to query
+    Returns:
+        result = result of dc query
+    """
     query_str = """
     SELECT DISTINCT ?drug ?id
     WHERE {{
     ?drug typeOf ChemicalCompound .
-    ?drug humanMetabolomeDatabaseID ?id .
-    ?drug humanMetabolomeDatabaseID {0} .
+    ?drug drugBankMetaboliteID ?id .
+    ?drug drugBankMetaboliteID {0} .
     }}
-    """.format(arr_hmdb_2)
+    """.format(arr_drug)
     result = dc.query(query_str)
-    dcid_hmdb = clean_result(result)
-    df = add_query_result(df, "hmdb", dcid_hmdb)
+    return result
+
+def main():
+    file_input = sys.argv[1]
+    file_output = sys.argv[2]
+    file_hmdb = sys.argv[3]
+
+    df = pd.read_csv(file_input, sep='\t')
+    dfh = pd.read_csv('hmdb.csv')
+
+    #inchikey matches with dc 
+    list_inchi = df[['inchiKey']].T.stack().tolist()
+    list_inchi_1 = list_inchi[1:1000]
+    list_inchi_2 = list_inchi[1000:2000]
+    list_inchi_3 = list_inchi[2000:2982]
+    arr_inchi_1 = np.array(list_inchi_1)
+    arr_inchi_2 = np.array(list_inchi_2)
+    arr_inchi_3 = np.array(list_inchi_3)
+    arr_inchi_list = list(arr_inchi_1, arr_inchi_2,arr_inchi_3 )
```

Comment:
This is quite error prone since it's easy to get the suffixes mixed up. It's also best not to hard-code the size of the list (e.g. 2000:2982)

Something like this could work for this chunk and the others below:

```python
def shard(list_to_shard, shard_size):
  sharded_list = []
   for i in range(0, len(list_to_shard, shard_size):
     shard = list_to_shard[i, i + shard_size]
     arr = np.array(shard)
     sharded_list.append(arr)
  return sharded_list
```

Reply:
thank you so much for the suggestion, added the shard function and it works like a charm!